### PR TITLE
EWS: Migrate needs notifications field from v1

### PR DESCRIPTION
**Pull request by kkrampa**
@gcapalbo

### CODE REVIEW

**Comment by kkrampa** [comment]
@czue Could you kick build again?

**Comment by czue** [comment]
done

File: custom/ewsghana/tasks.py [hunk 1]
```diff
@@ -198,18 +200,30 @@ def migrate_email_settings(domain):
...
-            ews_api.sms_user_sync(sms_user)
-        offset += 100
-        _, smsusers = endpoint.get_smsusers(filters={'with_more_than_one_number': True}, offset=offset)
+    domain_part = "%s.commcarehq.org" % domain
+
+    for sms_user in iterate_over_api_objects(endpoint.get_smsusers, filters={'needs_reminders': False}):
+        username_part = "%s%d" % (sms_user.name.strip().replace(' ', '.').lower(), sms_user.id)
+        username = "%s@%s" % (username_part[:(128 - (len(domain_part) + 1))], domain_part)
+        couch_user = CommCareUser.get_by_username(username)
+        if couch_user and 'needs_reminders' not in couch_user.user_data:
```
Comment (gcapalbo) [inline custom/ewsghana/tasks.py]: I think a better way to check would be `couch_user.user_data.get('needs_reminders') != 'False'`

File: custom/ewsghana/tasks.py [hunk 2]
```diff
@@ -198,18 +200,30 @@ def migrate_email_settings(domain):
...
+
+    to_save = []
+    for user in iter_docs(CommCareUser.get_db(), ids):
+        if 'needs_reminders' not in user['user_data']:
+            user['user_data']['needs_reminders'] = "True"
+            to_save.append(user)
+
+            if len(to_save) > 500:
+                CommCareUser.get_db().bulk_save(to_save)
+                to_save = []
```
Comment (gcapalbo) [inline custom/ewsghana/tasks.py]: You'll need to do a final bulk save outside the for loop as well if `len(to_save) > 0`

File: custom/ewsghana/tasks.py [hunk 3]
```diff
@@ -198,18 +200,30 @@ def migrate_email_settings(domain):
...
-    while smsusers:
-        for sms_user in smsusers:
-            ews_api.sms_user_sync(sms_user)
-        offset += 100
-        _, smsusers = endpoint.get_smsusers(filters={'with_more_than_one_number': True}, offset=offset)
+    domain_part = "%s.commcarehq.org" % domain
+
+    for sms_user in iterate_over_api_objects(endpoint.get_smsusers, filters={'needs_reminders': False}):
+        username_part = "%s%d" % (sms_user.name.strip().replace(' ', '.').lower(), sms_user.id)
+        username = "%s@%s" % (username_part[:(128 - (len(domain_part) + 1))], domain_part)
```
Comment (gcapalbo) [inline custom/ewsghana/tasks.py]: Is this the only way to look up the username / is there a better way to get the username to ensure we match?

File: custom/ewsghana/tasks.py [hunk 4]
```diff
@@ -198,18 +200,30 @@ def migrate_email_settings(domain):
...
-    offset = 0
-    _, smsusers = endpoint.get_smsusers(filters={'with_more_than_one_number': True})
-    while smsusers:
-        for sms_user in smsusers:
-            ews_api.sms_user_sync(sms_user)
-        offset += 100
-        _, smsusers = endpoint.get_smsusers(filters={'with_more_than_one_number': True}, offset=offset)
+    domain_part = "%s.commcarehq.org" % domain
+
+    for sms_user in iterate_over_api_objects(endpoint.get_smsusers, filters={'needs_reminders': False}):
```
Comment (gcapalbo) [inline custom/ewsghana/tasks.py]: Does this need to happen for web users too, or just sms users?

Comment (gcapalbo) [inline custom/ewsghana/tasks.py]: Can needs_reminders ever be null from the source?

Comment (kkrampa) [inline custom/ewsghana/tasks.py]: Only for sms users. In v1 it's not null field in db.

**Comment by kkrampa** [comment]
Now username is created in the same way as in migration.

**Pull request merged**